flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/DoubleAccumulator.java

```diff
@@ -0,0 +1,260 @@
+			return new DoubleMinimum(init);
+		}
+
+		public static DoubleMinimumFactory get() {
+			return INSTANCE;
+		}
+	}
+
+	/**
+	 * Factory for {@link DoubleSum}.
+	 */
+	final class DoubleSumFactory implements DoubleAccumulatorFactory<DoubleSum> {
+		private static final DoubleSumFactory INSTANCE = new DoubleSumFactory();
+
+		private DoubleSumFactory(){
+		}
+
+		@Override
+		public DoubleSum get(double init) {
+			return new DoubleSum(init);
+		}
+
+		public static DoubleSumFactory get() {
+			return INSTANCE;
+		}
+	}
+
+	/**
+	 * Factory for {@link DoubleAverage}.
+	 */
+	final class DoubleAverageFactory implements DoubleAccumulatorFactory<DoubleAverage> {
+		private static final DoubleAverageFactory INSTANCE = new DoubleAverageFactory();
+
+		private DoubleAverageFactory(){
+		}
+
+		@Override
+		public DoubleAverage get(double init) {
+			return new DoubleAverage(init);
+		}
+
+		public static DoubleAverageFactory get() {
+			return INSTANCE;
+		}
+	}
+
+	/**
+	 * {@link DoubleAccumulator} that returns the maximum value.
+	 */
+	final class DoubleMaximum implements DoubleAccumulator {
+
+		public static final String NAME = "max";
+
+		private double value;
+
+		private DoubleMaximum(double init) {
+			value = init;
+		}
+
+		@Override
+		public void add(double value) {
+			this.value = Math.max(this.value, value);
+		}
+
+		@Override
+		public double getValue() {
+			return value;
+		}
+
+		@Override
+		public String getName() {
+			return NAME;
+		}
+	}
+
+	/**
+	 * {@link DoubleAccumulator} that returns the minimum value.
+	 */
+	final class DoubleMinimum implements DoubleAccumulator {
+
+		public static final String NAME = "min";
+
+		private double value;
+
+		private DoubleMinimum(double init) {
+			value = init;
+		}
+
+		@Override
+		public void add(double value) {
+			this.value = Math.min(this.value, value);
+		}
+
+		@Override
+		public double getValue() {
+			return value;
+		}
+
+		@Override
+		public String getName() {
+			return NAME;
+		}
+	}
+
+	/**
+	 * {@link DoubleAccumulator} that returns the sum of all values.
+	 */
+	final class DoubleSum implements DoubleAccumulator {
+
+		public static final String NAME = "sum";
+
+		private double value;
+
+		private DoubleSum(double init) {
+			value = init;
+		}
+
+		@Override
+		public void add(double value) {
+			this.value += value;
+		}
+
+		@Override
+		public double getValue() {
+			return value;
+		}
+
+		@Override
+		public String getName() {
+			return NAME;
+		}
+	}
+
+	/**
+	 * {@link DoubleAccumulator} that returns the average over all values.
+	 */
+	final class DoubleAverage implements DoubleAccumulator {
+
+		public static final String NAME = "avg";
+
+		private double sum;
+		private int count;
+
+		private DoubleAverage(double init) {
+			sum = init;
+			count = 1;
+		}
+
+		@Override
+		public void add(double value) {
+			this.sum += value;
+			this.count++;
+		}
+
+		@Override
+		public double getValue() {
+			if (count == 0) {
```

Comment:
`count` cannot be `0` right? Because we set it to `1` in the constructor.

Reply:
yes, relic from a previous version where no initial value was set.